lib/Event/ReorderEvent.php

```diff
@@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
      */
     public function __construct($document, $destId, $after)
     {
-        parent::__construct($document);
+        $this->document = $document;
```

Comment:
To me it makes no sense to have interfaces for events. When I say
"polymorphic" I mean they are not intended to be replaced by the user by
a different implementation nor should this ever happen.

If there were a requirement to change this architecture lets think about
that when we meet it. I can see no reason to refactor this just for the
sake of refactoring to a solution which may or not be better than the
current solution.

On Tue, Jul 21, 2015 at 01:12:19AM -0700, Daniel Rotter wrote:
>    In [1]lib/Event/ReorderEvent.php:
> 
>  > @@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
>  >       */
>  >      public function __construct($document, $destId, $after)
>  >      {
>  > -        parent::__construct($document);
>  > +        $this->document = $document;
> 
>    Actually you are using the polymorphic, since you have type hinted the
>    handleHydrate method parameter with AbstractMappingEvent sometimes, so
>    that you can pass a PersistEvent in there. Therefore it would totally make
>    sense to me, to introduce these traits and interfaces, and I guess I will
>    do that in one of the next iterations. Would make some code reuse between
>    these elements also easier, e.g. there is no need to implement setNode
>    multiple times.
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu-document-manager/pull/28#discussion_r35078423
>    2. https://github.com/sulu-io/sulu-document-manager/pull/28/files#r35078423
>    3. https://github.com/sulu-io/sulu-document-manager/pull/28/files#r35078423